examples/calibration/plot_compare_calibration.py

```diff
@@ -4,119 +4,177 @@
-  this effect most strongly with random forests because the base-level trees
-  trained with random forests have relatively high variance due to feature
-  subsetting." As a result, the calibration curve shows a characteristic
-  sigmoid shape, indicating that the classifier could trust its "intuition"
-  more and return probabilities closer to 0 or 1 typically.
-
-* Support Vector Classification (SVC) shows an even more sigmoid curve as
-  the  RandomForestClassifier, which is typical for maximum-margin methods
-  (compare Niculescu-Mizil and Caruana [1]_), which focus on hard samples
-  that are close to the decision boundary (the support vectors).
-
-.. topic:: References:
-
-    .. [1] Predicting Good Probabilities with Supervised Learning,
-          A. Niculescu-Mizil & R. Caruana, ICML 2005
+of :term:`predict_proba` can be directly interpreted as a confidence level.
+For instance, a well calibrated (binary) classifier should classify the samples
+such that for the samples to which it gave a :term:`predict_proba` value close
+to 0.8, approximately 80% actually belong to the positive class.
+
+In this example we will compare the calibration of four different
+models, :ref:`Logistic_regression`, :ref:`gaussian_naive_bayes`,
+:ref:`Random Forest Classifier <forest>` and :ref:`Linear SVM
+<svm_classification>`.
 """
-print(__doc__)
+
+# %%
 
 # Author: Jan Hendrik Metzen <jhm@informatik.uni-bremen.de>
 # License: BSD Style.
 
-import numpy as np
-np.random.seed(0)
-
 import matplotlib.pyplot as plt
+from matplotlib.gridspec import GridSpec
+import numpy as np
 
-from sklearn import datasets
+from sklearn.datasets import make_classification
 from sklearn.naive_bayes import GaussianNB
 from sklearn.linear_model import LogisticRegression
 from sklearn.ensemble import RandomForestClassifier
 from sklearn.svm import LinearSVC
-from sklearn.calibration import calibration_curve
-
-X, y = datasets.make_classification(n_samples=100000, n_features=20,
-                                    n_informative=2, n_redundant=2)
+from sklearn.calibration import CalibrationDisplay
+
+# %%
+# Dataset
+# -------
+#
+# We will use a synthetic binary classification dataset with 100,000 samples
+# and 20 features. Of the 20 features, only 2 are informative, 2 are
+# redundant (random combinations of the informative features) and the
+# remaining 16 are uninformative (random numbers). Of the 100,000 samples,
+# 100 will be used for model fitting and the remaining for testing.
+
+X, y = make_classification(
+  n_samples=100_000, n_features=20, n_informative=2, n_redundant=2,
+  random_state=42
+)
 
 train_samples = 100  # Samples used for training the models
-
 X_train = X[:train_samples]
 X_test = X[train_samples:]
 y_train = y[:train_samples]
 y_test = y[train_samples:]
 
+# %%
+# Calibration curves
+# ------------------
+#
+# :class:`~sklearn.linear_model.LogisticRegression` returns well calibrated
```

Comment:
I would put the analysis after the graph. Indeed, we can make only a very small introduction by stating that we create our different models and create a cell. Then, mentioned that we will plot the calibration curve for each model to be able to compare them and in addition we will check the histogram of the mean predicted probability (and state the purpose of doing so).